packages/peregrine/lib/talons/MyAccount/useNewsletterSubscription.js

```diff
@@ -0,0 +1,73 @@
+import { useCallback, useState } from 'react';
+import { useMutation } from '@apollo/react-hooks';
+import { useToasts } from '@magento/peregrine';
+import { useUserContext } from '../../context/user';
+import { useAwaitQuery } from '../../hooks/useAwaitQuery';
+
+export const useNewsletterSubscription = props => {
+    const { subscribeMutation, customerQuery } = props;
+    const [{ currentUser, isSignedIn }, { getUserDetails }] = useUserContext();
+
+    const [subscribe, setSubscribe] = useState(
+        currentUser && currentUser.hasOwnProperty('is_subscribed')
+            ? currentUser.is_subscribed
+            : false
+    );
+    const [isSubmitting, setIsSubmitting] = useState(false);
+
+    const [subscription, { error: subscriptionError }] = useMutation(
+        subscribeMutation,
+        {
+            fetchPolicy: 'no-cache'
+        }
+    );
```

Comment:
Mutations are no-cache by default, that shouldn't be needed. Also an example of using mutation state to track submit state.
```suggestion
    const [subscription, { error: subscriptionError, loading: isSubmitting }] = useMutation(
        subscribeMutation
    );
```